server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -443,6 +443,13 @@ public ClusterState applyCreateIndexRequest(
 
             // The backing index may have a different name or prefix than the data stream name.
             final String name = request.dataStreamName() != null ? request.dataStreamName() : request.index();
+            // Do not apply any templates to system indices
+            // Using applyCreateIndexRequestWithV1Templates with empty list instead of applyCreateIndexRequestWithV2Template with null
+            // template as applyCreateIndexRequestWithV2Template has assertions when template is null
+            if (systemIndices.isSystemIndex(name)) {
```

Comment:
Let's move the determination of `name` and this new block of code above the `isHiddenFromRequest` assignment since it won't be needed.